sdk/python/tests/compiler/compiler_withparams_test_helper.py

```diff
@@ -0,0 +1,97 @@
+"""This module is useful for generating yaml files for the withParams tests and for running unformal
+compiler tests during development."""
+import time
+
+from kfp.compiler import compiler
+from kfp import dsl
+from kfp.dsl import _for_loop
+
+
+class Coder:
+    def __init__(self, ):
+        self._code_id = 0
+
+    def get_code(self, ):
+        self._code_id += 1
+        return '{code:0{num_chars:}d}'.format(code=self._code_id, num_chars=_for_loop.LoopArguments.NUM_CODE_CHARS)
+
+
+dsl.ParallelFor._get_unique_id_code = Coder().get_code
+
+
+if __name__ == '__main__':
+    do_output = True
+
+    params = {}
+    if do_output:
+        @dsl.pipeline(name='my-pipeline')
+        def pipeline():
+            op0 = dsl.ContainerOp(
+                name="my-out-cop0",
+                image='python:alpine3.6',
+                command=["sh", "-c"],
```

Comment:
BTW, Why do you run shell instead of just running python? You'll need less escaping.

Reply:
it's just historical.